hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/FieldProbe.java

```diff
@@ -71,14 +75,14 @@ private String getName(String namePrefix) {
             throw new IllegalArgumentException(format("@Probe field '%s' is of an unhandled type", field));
         }
 
-        if (AccessibleObjectProbe.isDouble(type)) {
+        if (isDouble(type)) {
             return new DoubleFieldProbe<S>(field, probe, type);
```

Comment:
The factory has knowledge about the subclasses. Why do you think that is a problem?